sonar-db/src/main/java/org/sonar/db/charset/MssqlCharsetHandler.java

```diff
@@ -46,97 +42,81 @@
   private static final String BIN = "BIN";
   private static final String BIN2 = "BIN2";
 
-  protected MssqlCharsetHandler(SqlExecutor selectExecutor) {
+  private final Metadata metadata;
+
+  MssqlCharsetHandler(SqlExecutor selectExecutor) {
     super(selectExecutor);
+    this.metadata = new Metadata();
+  }
+
+  @VisibleForTesting
+  MssqlCharsetHandler(SqlExecutor selectExecutor, Metadata metadata) {
+    super(selectExecutor);
+    this.metadata = metadata;
   }
 
   @Override
-  void handle(Connection connection, Set<DatabaseCharsetChecker.Flag> flags) throws SQLException {
-    logInit(flags);
+  void handle(Connection connection, DatabaseCharsetChecker.State state) throws SQLException {
+    if (state == DatabaseCharsetChecker.State.FRESH_INSTALL) {
```

Comment:
I would use a switch on the enum:
* would clearly show that nothing is done on `STARTUP`
* default would raise an error and force any new value in `State` to be dealt with